src/samplers/remote_sampler.js

```diff
@@ -101,38 +101,54 @@ export default class RemoteControlledSampler {
 
     _refreshSamplingStrategy() {
         let serviceName: string = encodeURIComponent(this._serviceName);
-        let url: string = `http://${this._host}:${this._port}/sampling?service=${serviceName}`;
-        request.get(url, (err, response) => {
-            if (err) {
-                this._logger.error(`Error in fetching sampling strategy from ${url}: ${err}.`);
-                this._metrics.samplerQueryFailure.increment(1);
-                return;
-            }
-            this._metrics.samplerRetrieved.increment(1);
-            let strategy;
-            try {
-                strategy = JSON.parse(response.body);
-                if (!strategy) {
-                    throw 'Malformed response: ' + response.body;
-                }
-            } catch (error) {
-                this._logger.error(`Error in parsing sampling strategy: ${error}.`);
-                this._metrics.samplerParsingFailure.increment(1);
-                return;
-            }
-            try {
-                if (this._updateSampler(strategy)) {
-                    this._metrics.samplerUpdated.increment(1);
-                }
-            } catch (error) {
-                this._logger.error(`Error in updating sampler: ${error}.`);
-                this._metrics.samplerParsingFailure.increment(1);
-                return;
+        http.get({
+            'host': this._host,
+            'port': this._port,
+            'path': `/sampling?service=${serviceName}`
+        }, (res) => {
+            // explicitly treat incoming data as utf8 (avoids issues with multi-byte chars)
+            res.setEncoding('utf8');
+
+            // incrementally capture the incoming response body
+            let body = '';
```

Comment:
Considering that there's even a custom module for that (https://github.com/raypulver/string-buffer), I am guessing there isn't a std string buffer. There's Buffer, but it works on bytes.

Given the frequency of polling for the sampling rate I don't think the gc pressure matters here.